contracts/lbp/LBPWrapper.sol

```diff
@@ -0,0 +1,145 @@
+/*
+██████╗░██████╗░██╗███╗░░░███╗███████╗██████╗░░█████╗░░█████╗░
+██╔══██╗██╔══██╗██║████╗░████║██╔════╝██╔══██╗██╔══██╗██╔══██╗
+██████╔╝██████╔╝██║██╔████╔██║█████╗░░██║░░██║███████║██║░░██║
+██╔═══╝░██╔══██╗██║██║╚██╔╝██║██╔══╝░░██║░░██║██╔══██║██║░░██║
+██║░░░░░██║░░██║██║██║░╚═╝░██║███████╗██████╔╝██║░░██║╚█████╔╝
+╚═╝░░░░░╚═╝░░╚═╝╚═╝╚═╝░░░░░╚═╝╚══════╝╚═════╝░╚═╝░░╚═╝░╚════╝░
+*/
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// solium-disable linebreak-style
+pragma solidity ^0.8.4;
+
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../utils/interface/ILBPFactory.sol";
+import "../utils/interface/IVault.sol";
+import "../utils/interface/ILBP.sol";
+import "hardhat/console.sol";
+
+
+contract LBPWrapper {
+
+    address public owner;
+    address public LBPFactory;
+    address public lbp;
+    uint256 public swapFeePercentage;
+    bool isInitialized;
+
+    modifier onlyOwner{
+        require(msg.sender == owner, "LBPWrapper: only owner function");
+        _;
+    }
+
+    /**
+     * @dev              transfer ownership to new owner
+     * @param _newOwner  new owner address
+     */
+    function transferOwnership(address _newOwner) public onlyOwner{
+        require(_newOwner != address(0), "LBPWrapper: new owner cannot be zero");
+        owner = _newOwner;
+    }
+
+    /**
+     * @dev                       initialize lbp wrapper contract
+     * @param _LBPFactory         LBP factory address
+     * @param _swapFeePercentage  swap fee percentage
+     */
+    function initialize (
+            address _LBPFactory,
+            uint256 _swapFeePercentage
+    ) public
+    {
+        owner = msg.sender;
+        LBPFactory = _LBPFactory;
```

Comment:
I think the LBPFactory we're going to use is a contract already launched by Balancer. If I understand you correct then what we want to store is the interface to the contract, not the contract address itself. So that's why it would be better to store `ILBPFactory` into a state variable? 

Also, what do you mean by Initializing the contract immediately here? I'm guessing you mean when calling `LBPWrapper.initialize()` to have it do all that is done in `deployLbpFromFactory`?

Reply:
Basically the first paragraph of your comment is what I meant.

Reply:
No worries :) . So my first paragraph would be your suggestions then, right? Instead of sending the address of the contract, we store the interface to the contract. Mihir probably knows how to do this but for myself, would we do this by sending the address of an interface?

Reply:
Yes, I guess I would find that a cleaner solution

Reply:
```suggestion
        LBPFactory = ILBPFactory(_LBPFactory);
```

Reply:
```suggestion
        LBPFactory = ILBPFactory(_LBPFactory);
```

Reply:
Implemented it. Thanks for your help